WooCommerce/Classes/ServiceLocator/PushNotification.swift

```diff
@@ -42,7 +50,30 @@ extension PushNotification {
               }
         let subtitle = alert.string(forKey: APNSKey.alertSubtitle)
         let message = alert.string(forKey: APNSKey.alertMessage)
-        return PushNotification(noteID: noteID, siteID: siteID, kind: noteKind, title: title, subtitle: subtitle, message: message)
+        let note: Note? = noteFromNoteData(userInfo.string(forKey: APNSKey.noteFullData))
+        return PushNotification(noteID: noteID, siteID: siteID, kind: noteKind, title: title, subtitle: subtitle, message: message, note: note)
+    }
+
+    static func noteFromNoteData(_ noteFulldata: String?) -> Note? {
```

Comment:
Do you think that `NoteFromCompressedData` could make more sense to the reader?
Also, adding some comments on the method signature could help future readers.